### PR TITLE
More and better examples and documentation

### PR DESCRIPTION
**Pull request by sebastianludwig**
Take a look and feel free to amend the README commit as you please.

### CODE REVIEW

File: Example/LDOLayoutTemplates/ComplexViewController.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    @IBOutlet weak var portraitLargeEventsLayout: LDOLayoutTemplate!
+    @IBOutlet weak var landscapeLargeEventsLayout: LDOLayoutTemplate!
+    private var defaultLayout: LDOLayoutTemplate!
+    private var largeBlueLayoutActive = false
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        defaultLayout = LDOLayoutTemplate.forCurrentState(basedOnTemplate: portraitLargeEventsLayout)
+        assert(defaultLayout != nil)
```
Comment (jlnr) [inline Example/LDOLayoutTemplates/ComplexViewController.swift]: Why does this method return an optional/nullable value?

File: Example/LDOLayoutTemplates/ComplexViewController.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        
+        if !largeBlueLayoutActive {
+            return
+        }
+        
+        coordinator.animate(alongsideTransition: { _ in
+            if size.width > size.height {
+                self.landscapeLargeEventsLayout.apply()
```
Comment (jlnr) [inline Example/LDOLayoutTemplates/ComplexViewController.swift]: `largeBlueLayout` <-> `{portrait,landscape}LargeEventsLayout` is inconsistent.

File: Example/LDOLayoutTemplates/ViewController.m [hunk 3]
```diff
@@ -40,9 +40,10 @@ - (void)applyLayoutTemplate:(LDOLayoutTemplate *)template
 
 - (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
 {
-    if (size.width > size.height) {
-        [self applySecond];
-    }
+    [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+    [coordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext>  context) {
```
Comment (jlnr) [inline Example/LDOLayoutTemplates/ViewController.m]: One space too much 👀

File: Example/LDOLayoutTemplates/ComplexViewController.swift [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                self.defaultLayout.apply()
+            } else {
+                if self.view.bounds.width > self.view.bounds.height {
+                    self.landscapeLargeEventsLayout.apply()
+                } else {
+                    self.portraitLargeEventsLayout.apply()
+                }
+            }
+            self.view.layoutIfNeeded()
+            self.largeBlueLayoutActive = !self.largeBlueLayoutActive
```
Comment (jlnr) [inline Example/LDOLayoutTemplates/ComplexViewController.swift]: `.toggle()`

File: README.md [hunk 5]
```diff
@@ -1,22 +1,76 @@
...
+                self.defaultLayout.apply()
+            } else {
+                if self.view.bounds.width > self.view.bounds.height {
+                    self.landscapeLargeEventsLayout.apply()
+                } else {
+                    self.portraitLargeEventsLayout.apply()
+                }
+            }
+            self.view.layoutIfNeeded()
+            self.largeBlueLayoutActive = !self.largeBlueLayoutActive
```
Comment (jlnr) [inline README.md]: `.toggle()`

File: README.md [hunk 6]
```diff
@@ -1,22 +1,76 @@
...
+            if size.width > size.height {
+                self.landscapeLargeEventsLayout.apply()
+            } else {
+                self.portraitLargeEventsLayout.apply()
+            }
+            self.view.layoutIfNeeded()
+        }, completion: nil)
+    }
+    
+    @IBAction func toggleSize() {
```
Comment (jlnr) [inline README.md]: I think this should also be renamed to match `largeBlueLayoutActive`.

File: README.md [hunk 7]
```diff
@@ -1,22 +1,76 @@
...
+![Demo](Screenshots/ComplexExample.gif)
+
+### With this
+
+```Swift
+class ComplexViewController: UIViewController {
+    @IBOutlet weak var portraitLargeEventsLayout: LDOLayoutTemplate!
+    @IBOutlet weak var landscapeLargeEventsLayout: LDOLayoutTemplate!
+    private var defaultLayout: LDOLayoutTemplate!
+    private var largeBlueLayoutActive = false
```
Comment (jlnr) [inline README.md]: What do you think about moving the transition from the `IBAction` to a `didSet` clause here?

**Review by jlnr — CHANGES_REQUESTED** [review]
Some code-level suggestions. Will push a README commit later.

File: README.md (hunk 7 above)
Comment (sebastianludwig) [reply README.md]: Plus: the `largeListLayoutActive` bool and the applied layout are more in sync
Minus: the code is more spread out and fancier
Neutral: I dislike heavy `didSet`s and changing the layout is at least close to that category

File: Example/LDOLayoutTemplates/ComplexViewController.swift [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            if size.width > size.height {
+                self.landscapeLargeEventsLayout.apply()
+            } else {
+                self.portraitLargeEventsLayout.apply()
+            }
+            self.view.layoutIfNeeded()
+        }, completion: nil)
+    }
+    
+    @IBAction func toggleLargeListLayout() {
```
Comment (jlnr) [inline Example/LDOLayoutTemplates/ComplexViewController.swift]: It's still largeEvents != largeList.

File: Example/LDOLayoutTemplates/ComplexViewController.swift [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    private var largeListLayoutActive = false
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        defaultLayout = LDOLayoutTemplate(forCurrentStateBasedOn: portraitLargeEventsLayout)
+    }
+    
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        guard largeListLayoutActive else { return }
```
Comment (jlnr) [inline Example/LDOLayoutTemplates/ComplexViewController.swift]: +Comments for the `guard` and the transition below? Might help when users try to understand the source code without looking at the app at the very same time.

File: LDOLayoutTemplates/Classes/LDOLayoutTemplate.h [hunk 10]
```diff
@@ -10,11 +10,11 @@
 
 @interface LDOLayoutTemplate : NSObject
 
-@property (nonatomic, weak) IBOutlet UIView *destinationView;
-@property (nonatomic) IBOutlet UIView *templateView;
+@property (nullable, nonatomic, weak) IBOutlet UIView *destinationView;
```
Comment (jlnr) [inline LDOLayoutTemplates/Classes/LDOLayoutTemplate.h:13]: Do we want to use `nullable` here? How do we deal with missing views?

Would it be slightly better (for edge cases) to use `strong`?

**Review by jlnr — CHANGES_REQUESTED** [review]
More things and thoughts. Will send README suggestions in a separate PR after merging this one.

**Comment by jlnr** [comment]
LGTM

**Pull request merged**